cpp/include/cudf/search.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "cudf.h"
+#include "types.hpp"
+#include <vector>
+
+namespace cudf {
+
+/**---------------------------------------------------------------------------*
+ * @brief Find smallest indices in sorted column where values should be 
+ *  inserted to maintain order
+ * 
```

Comment:
Maybe insert a more explicit explanation here?

"For each value v in @p values, the corresponding output is the element of @p column with smallest index that does not compare less than v."